src/herbpy/action/pushing.py

```diff
@@ -0,0 +1,83 @@
+    @param table The table kinbody the object is resting on
+    @param goal_position The desired (x,y) position of the object in world coordinates
+    @param goal_radius The max distance from goal_position for the object to be to 
+      still consider the goal achieved
+    @param manip The manipulator to use for the push - if None the active manipulator is used
+    @param max_plan_duration The max time to run the planner
+    @param shortcut_time The amount of time to spend shortcutting, if 0. no shortcutting is performed
+    """
+    # Get a push planner
+    from or_pushing.push_planner import PushPlanner
+    planner = PushPlanner(robot.GetEnv())
+
+    # Get the manipulator
+    if manip is None:
+        with robot.GetEnv():
+            manip = robot.GetActiveManipulator()
+
+    # Make the state bounds be at the edges of the table
+    with robot.GetEnv():
+        from prpy.rave import Disabled
+        from prpy.util import ComputeEnabledAABB
+        with Disabled(table, padding_only=True):
+            table_aabb = ComputeEnabledAABB(table)
+    table_pos = table_aabb.pos()
+    table_extents = table_aabb.extents()
+
+    sbounds = {'high': [table_pos[0] + table_extents[0],
+                        table_pos[1] + table_extents[1],
+                        2.*numpy.pi],
+               'low': [table_pos[0] - table_extents[0],
+                       table_pos[1] - table_extents[1],
+                       0]}
+    
+    # Get the current pose of the manipulator and assume we want to keep that pose throughout
+    #  the push
+    with robot.GetEnv():
+        ee_pushing_transform = manip.GetEndEffectorTransform()
+    ee_pushing_transform[:2,3] = [0., 0.] #ignore x,y pose
+
+    # Compute the goal pose
+    with robot.GetEnv():
+        goal_pose = obj.GetTransform()
+    table_height = table_pos[2] + table_extents[2] + 0.01
+    goal_pose[:3,3] = [goal_position[0],
+                       goal_position[1],
+                       table_height]
+
+    with robot.CreateRobotStateSaver():
+        traj = planner.PushToPose(robot, obj, goal_pose,
+                                  state_bounds = sbounds,
+                                  pushing_manifold = ee_pushing_transform.flatten().tolist(),
+                                  max_plan_duration = max_plan_duration,
+                                  goal_epsilon = goal_radius,
+                                  **kw_args)
+    if traj is None:
+        raise PlanningError('Failed to find pushing plan')
+
+    # Execute
+    from prpy.viz import RenderTrajectory
+    with RenderTrajectory(robot, traj, color=[1, 0, 0, 1]):
+        if shortcut_time > 0:
+            traj = planner.ShortcutPath(timelimit=shortcut_time)
+        with RenderTrajectory(robot, traj, color=[0, 0, 1, 1]):
+          if manip.simulated:
+              planner.ExecutePlannedPath()
+          else:
+              robot.ExecuteTrajectory(traj)
+              planner.SetFinalObjectPoses()
```

Comment:
We could. ```ExecutePlannedPath``` on the planner renders the object movement as well as the arm movement, basically by subsampling states (containing robot and object pose) on the original OMPL trajectory and stepping the environment through these states.  ```ExecuteTrajectory``` will just execute an arm trajectory as normal, and move the arm through all the objects. Then ```SetFinalObjectPose``` teleports the objects to their final planned locations at the end of the trajectory. I think this is our only option on the real robot, but the ```ExecutePlannedPath``` option looks a little nicer in simulation.